developer-experience/readme.md

```diff
@@ -0,0 +1,203 @@
+
+### The F5 Contract
+
+The F5 contract aims for the ability to run the end-to-end solution with the following steps.
+
+1. Clone - git clone [`my-repo-url-here`]
+2. Configure - set any configuration values that need to be unique to the individual (i.e. update a .env file)
+3. Press F5 - launch the solution with debugging attached.
+
+Most IDE's have some form of a task runner that can be used to automate the build, execute, and attach steps. Try to leverage these such that the steps can all be run with as few manual steps as possible.
+
+### DevEx Champion Actively Seek Improvements
+
+The DevEx champion should actively seek areas where the team has opportunity to improve. For example, do they need to deploy their changes to an environment off their laptop before they can validate if what they did worked. Rather than debugging locally, do they have to do this repetitively to get to a working solution? Does this take several minutes each iteration? Does this block other developers due to the contention on the environment?
+
+The following are ceremonies that the DevEx champion can use to find potential opportunities
+
+- Retrospectives. Is feedback being raised that relates to the essential tasks being difficult or unwieldy?
+- Standup Blockers. Are individuals getting blocked or stumbling on the essential tasks?
+
+As opportunities are identified, the DevEx champion can translate these into actionable stories for the product backlog.
+
+### Single IDE Instance
+
+Avoid requiring multiple instances of the IDE to running concurrently to run and debug the solution locally. This will negatively impact the above measure. Additionally, that negative impact is multiplied by each instance required.
+
+- Configuration steps * IDE instances
+- Build steps * IDE instances
+- Start/Debug steps * IDE instances
+- Stop steps * IDE instances
+- Run test steps * IDE instances
+- Documenting all of the above * IDE instances
+
+### Minimize the Number of Repositories
+
+Splitting a solution across multiple repositories can negatively impact the above measures. This can also negatively impact other areas such as Pull Requests, Automated Testing, Continuous Integration, and Continuous Delivery. Similar to the IDE instances, the negative impact is multiplied by the number of repositories.
+
+- clone steps * # of repos
+- branching steps * # of repos
+- commit steps * # of repos
+- CI * # of repos
+- Pull request reviews & merges * # of repos
+
+#### Atomic Pull Requests
+
+When the solution is encapsulated within a single repository, it also allows pull requests to represent a change across multiple layers. This is especially helpful when a change requires changes to a shared contract between multiple components. For example, a story requires that an api endpoint is changed. With this strategy the api and web client could be updated with the same pull request. This avoids master being broken temporarily while waiting on dependent pull requests to merge.

```

Comment:
Nice

Reply:
One of the many benefits when we consolidated repos on my last project ;)